src/backend/TrafficCourts/Workflow.Service/Consumers/SendEmailToDisputantConsumer.cs

```diff
@@ -31,46 +32,62 @@ public SendEmailToDisputantConsumer(IEmailSenderService emailSenderService, IClo
     public async Task Consume(ConsumeContext<SendDisputantEmail> context)
     {
         using var scope = _logger.BeginConsumeScope(context, message => message.NoticeOfDisputeGuid );
+        _logger.LogDebug("Calling email sender service");
 
-        Dispute? dispute = await _oracleDataApiService.GetDisputeByNoticeOfDisputeGuidAsync(NoticeOfDisputeGuid: context.Message.NoticeOfDisputeGuid, context.CancellationToken);
+        try
+        {
+            // search for dispute log warning if not found
+            Dispute? dispute = await _oracleDataApiService.GetDisputeByNoticeOfDisputeGuidAsync(NoticeOfDisputeGuid: context.Message.NoticeOfDisputeGuid, context.CancellationToken);
+            if (dispute is null) { 
+                _logger.LogWarning("Sending email without existing dispute", context.Message); 
+            }
 
-        EmailMessage emailMessage = context.Message.Message;
+            EmailMessage emailMessage = context.Message.Message;
 
-        // try to fill in to if missing
-        if (emailMessage.To.IsNullOrEmpty() && dispute is not null && dispute.EmailAddress is not null && dispute.EmailAddressVerified == true)
-            emailMessage.To = dispute.EmailAddress;
+            // try to fill in to if missing
+            if (emailMessage.To.IsNullOrEmpty() && dispute is not null && dispute.EmailAddress is not null && dispute.EmailAddressVerified == true)
+                emailMessage.To = dispute.EmailAddress;
 
-        _logger.LogDebug("Calling email sender service");
+            var result = emailMessage.To.IsNullOrEmpty() ? SendEmailResult.Filtered : await _emailSenderService.SendEmailAsync(emailMessage, context.CancellationToken);
```

Comment:
This `IsNullOrEmpty()` is from `Microsoft.IdentityModel.Tokens` which checks whether enumerable if character is null or empty. Typically, should be using `string.IsNullOrEmpty(emailMessage.To)`